File: services/spring/inventory_service/src/main/java/com/smartfulfillment/inventory_service/entity/Product.java
package com.smartfulfillment.inventory_service.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "product_id", updatable = false, nullable = false)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String sku;     // Stock Keeping Unit

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private BigDecimal price;

    private String imageUrl;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: services/go/inventory-service-twin/internal/models/product.go
package models

import (
	"time"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

// Product maps to Spring's Product entity and `products` table.
type Product struct {
	ID               uuid.UUID       `gorm:"type:uuid;primaryKey;column:product_id" json:"id"`
	SKU              string          `gorm:"not null;unique;size:50" json:"sku"`
	Name             string          `gorm:"not null;size:100" json:"name"`
	Description      string          `gorm:"type:text" json:"description,omitempty"`
	Price            decimal.Decimal `gorm:"not null;type:numeric(10,2)" json:"price"`
	ImageURL         string          `gorm:"column:image_url" json:"imageUrl,omitempty"`
	CreatedAt        time.Time       `gorm:"column:created_at;autoCreateTime" json:"createdAt"`
	LowStockThreshold int            `gorm:"column:low_stock_threshold;not null;default:10" json:"lowStockThreshold"`
}

func (Product) TableName() string {
	return "products"
}

